From de788a17ed4866d96c8dbb2b01b552d5ddb43ca4 Mon Sep 17 00:00:00 2001
From: Mike Zeller <mike@mikezeller.net>
Date: Thu, 13 Sep 2018 17:47:28 +0000
Subject: [PATCH] OS-7243 libvarpd_c_destroy gets away with pointer murder
 Reviewed by: Dan McDonald <danmcd@joyent.com> Reviewed by: Robert Mustacchi
 <rm@joyent.com> Approved by: Robert Mustacchi <rm@joyent.com>

---
 usr/src/lib/varpd/libvarpd/common/libvarpd_client.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c b/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c
index b0fa907386..260bbb6e92 100644
--- a/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c
+++ b/usr/src/lib/varpd/libvarpd/common/libvarpd_client.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright 2015 Joyent, Inc.
+ * Copyright 2018 Joyent, Inc.
  */
 
 /*
@@ -113,7 +113,7 @@ libvarpd_c_destroy(varpd_client_handle_t *chp)
 		libvarpd_panic("failed to close door fd %d: %d",
 		    client->vcl_doorfd, errno);
 
-	umem_free(chp, sizeof (varpd_client_handle_t *));
+	umem_free(chp, sizeof (varpd_client_t));
 }
 
 int
-- 
2.21.0

